FirebaseCore/Tests/Unit/Swift/FirebaseAppTests.swift

```diff
@@ -0,0 +1,382 @@
+    guard let defaultApp = FirebaseApp.app() else {
+      return XCTFail("Could not unwrap default app")
+    }
+    XCTAssertEqual(defaultApp.name, testApp.firebaseDefaultAppName)
+  }
+
+  func testGetNameOfApp() throws {
+    XCTAssertNil(FirebaseApp.app(name: testApp.testAppName1))
+
+    let options = FirebaseOptions(googleAppID: testOptions.googleAppID,
+                                  gcmSenderID: testOptions.gcmSenderID)
+    FirebaseApp.configure(name: testApp.testAppName1, options: options)
+
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let app = FirebaseApp.app(name: testApp.testAppName1) else {
+      return XCTFail("Could not unwrap app")
+    }
+    XCTAssertEqual(app.name, testApp.testAppName1)
+  }
+
+  func testOptionsForApp() {
+    FirebaseApp.configure()
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let defaultApp = FirebaseApp.app() else {
+      return XCTFail("Could not unwrap default app")
+    }
+    let defaultOptions = FirebaseOptions.defaultOptions()
+    XCTAssertEqual(defaultApp.options, defaultOptions)
+
+    let options = FirebaseOptions(googleAppID: testOptions.googleAppID,
+                                  gcmSenderID: testOptions.gcmSenderID)
+    let superSecretURLScheme = "com.supersecret.googledeeplinkurl"
+    options.deepLinkURLScheme = superSecretURLScheme
+    FirebaseApp.configure(name: testApp.testAppName1, options: options)
+
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let app = FirebaseApp.app(name: testApp.testAppName1) else {
+      return XCTFail("Could not unwrap app")
+    }
+    XCTAssertEqual(app.name, testApp.testAppName1)
+    XCTAssertEqual(app.options.googleAppID, testOptions.googleAppID)
+    XCTAssertEqual(app.options.gcmSenderID, testOptions.gcmSenderID)
+    XCTAssertEqual(app.options.deepLinkURLScheme, superSecretURLScheme)
+    XCTAssertNil(app.options.androidClientID)
+  }
+
+  func testFirebaseDataCollectionDefaultEnabled() {
+    FirebaseApp.configure()
+
+    // TODO: Use XCTUnwrap after dropping support for Xcode 10
+    guard let app = FirebaseApp.app() else {
+      return XCTFail("Could not unwrap default app")
+    }
+
+    // defaults to true unless otherwise set to no in app's Info.plist
+    XCTAssertTrue(app.isDataCollectionDefaultEnabled)
+
+    app.isDataCollectionDefaultEnabled = false
+    XCTAssertFalse(app.isDataCollectionDefaultEnabled)
+
+    // reset to defautl true since it will persist across runs of the app/tests
+    app.isDataCollectionDefaultEnabled = true
+  }
+
+  // MARK: - Helpers
+
+  private func expectedUserInfoForApp(named name: String, isDefaultApp: Bool) -> NSDictionary {
+    return [
+      "FIRAppNameKey": name,
+      "FIRAppIsDefaultAppKey": NSNumber(value: isDefaultApp),
+      "FIRGoogleAppIDKey": testOptions.googleAppID,
+    ]
+  }
+
+  private func expectAppConfigurationNotification(with expectedUserInfo: NSDictionary) {
```

Comment:
I feel like these two could be merged to make it less verbose and more clear:

```
private func expectAppConfigurationNotification(appName: String, isDefaultApp: Bool) {
}
```

Changes the call site from:
```
let expectedUserInfo = expectedUserInfoForApp(named: testApp.testAppName1,
                                                  isDefaultApp: false)
expectAppConfigurationNotification(with: expectedUserInfo)
```
to
```
expectAppConfigurationNotification(appName: testApp.testAppName1, isDefaultApp: false)
```

Reply:
good point- I like that a lot better! done